pkg/instancetype/instancetype.go

```diff
@@ -1337,9 +1342,14 @@ func applyInterfacePreferences(preferenceSpec *instancetypev1beta1.VirtualMachin
 		if preferenceSpec.Devices.PreferredInterfaceModel != "" && vmiIface.Model == "" {
 			vmiIface.Model = preferenceSpec.Devices.PreferredInterfaceModel
 		}
-		if preferenceSpec.Devices.PreferredInterfaceMasquerade != nil && isInterfaceBindingUnset(vmiIface) && isInterfaceOnPodNetwork(vmiIface.Name, vmiSpec) {
+
+		if preferenceSpec.Devices.PreferredInterfaceMasquerade != nil && isInterfaceBindingUnset(vmiIface) && isInterfaceOnRequiredNetwork(vmiIface.Name, vmiSpec, virtv1.PodNetwork{}) {
 			vmiIface.Masquerade = preferenceSpec.Devices.PreferredInterfaceMasquerade.DeepCopy()
 		}
+
+		if preferenceSpec.Devices.PreferredInterfaceSRIOV != nil && isInterfaceBindingUnset(vmiIface) && isInterfaceOnRequiredNetwork(vmiIface.Name, vmiSpec, virtv1.MultusNetwork{}) {
```

Comment:
Can we add a check to see if the multus network has SRIOV? Can you tell from the NetworkAttachmentDefinition is is referring to?

Reply:
`isInterfaceBindingUnset` checks if the interface has a binding set. If a value is present, SRIOV will not be applied. I think that is the check you are referring to, right?

Reply:
No, I want you to check that the multus network that is used is actually backed by SRIOV. You need to get the referred `NetworkAttachmentDefinition` and check if its `type` is `sriov`.

You can get a NAD e.g. like this: https://github.com/kubevirt/kubevirt/blob/17abe1a13f7b8649637690c69c1a155abc4975ee/pkg/virt-controller/services/renderresources.go#L494